### ISSUE TITLE
Dynamically get the column names and know their types

### ISSUE THREAD

**Issue by jaeggerr** [typeorm/typeorm#1160]
I am trying to build an API following the JSONAPI requirements.
http://jsonapi.org/format/#document-structure

Entities returned by endpoints should look like this:
```js
{
  "data": {
    "type": "articles",
    "id": "1",
    "attributes": {
      // ... this article's attributes
    },
    "relationships": {
      // ... this article's relationships
    }
  }
}
```

So I need to separate the attributes (primitive values such as strings, numbers, dates, ...) and the relationships to other entities.

Is there an way I could dynamically find all the columns names of a BaseEntity subclass and know if they are primitive columns or relationship, so that I could dynamically format my JSON response?
Ideally, I'd like to do it statically, without needing to create a new instance of that entity.

Thank you

**Comment by pleerock** [comment]
Yes you can use `connection.getMetadata(Article)` which returns you instance of `EntityMetadata` where you can find all information you need about your entity.
